programs/merkle-tree/src/config_accounts/register_verifier.rs

```diff
@@ -28,3 +28,19 @@ pub struct RegisterVerifier<'info> {
     pub system_program: Program<'info, System>,
     pub rent: Sysvar<'info, Rent>,
 }
+
+pub fn process_register_verifier(
+    ctx: Context<RegisterVerifier>,
+    verifier_pubkey: Pubkey,
+) -> Result<()> {
+    if !ctx
+        .accounts
+        .merkle_tree_authority_pda
+        .enable_permissionless_merkle_tree_registration
```

Comment:
registering verifiers should never be permissionless, otherwise anyone can just rug the protocol.
The check should be if signer is not authority throw error.

Reply:
Why is this part of the Merkle tree refactor scope anyway?